src/iexamine.cpp

```diff
@@ -1832,88 +1894,219 @@ void iexamine::aggie_plant(player &p, const tripoint &examp)
+                weed_info = _( "few" );
+            } else {
+                weed_info = _( "none" );
+            }
+
+            if( grow_ratio > 1.0 ) {
+                grow_info = _( "Harvest is ready" );
+            } else if( grow_ratio > 0.66 ) {
+                grow_info = _( "Harvest is almost ready" );
+            } else if( grow_ratio > 0.33 ) {
+                grow_info = _( "Growing is in the middle" );
+            } else {
+                grow_info = _( "Growing just started." );
+            }
+
+            if( health_ratio > 0.75 ) {
+                health_info = _( "Plants health is very good" );
+            } else if( health_ratio > 0.25 ) {
+                health_info = _( "Plants health is good" );
+            } else if( health_ratio > -0.25 ) {
+                health_info = _( "Plants health is average" );
+            } else if( health_ratio > -0.75 ) {
+                health_info = _( "Plants health is bad" );
+            } else {
+                health_info = _( "Plants health is very bad" );
+            }
         }
-        if (query_yn(_("Fertilize the %s"), pname.c_str() )) {
-        std::vector<itype_id> f_types;
-        std::vector<std::string> f_names;
-            for( auto &f : f_inv ) {
-                if( std::find( f_types.begin(), f_types.end(), f->typeId() ) == f_types.end() ) {
-                    f_types.push_back( f->typeId() );
-                    f_names.push_back( f->tname() );
-                }
+
+        pmenu.text = pname;
+        pmenu.addentry( 1, true,  MENU_AUTOASSIGN, _( "add water          %s" ), water_info );
+        pmenu.addentry( 2, true,  MENU_AUTOASSIGN, _( "add fertilizer     %s" ), fertilizer_info );
+        pmenu.addentry( 3, true,  MENU_AUTOASSIGN, _( "remove weeds       %s" ), weed_info );
+        if( p.get_skill_level( skill_survival ) >= 2 ) {
+            pmenu.addentry( 0, false, MENU_AUTOASSIGN, health_info );
+            pmenu.addentry( 0, false, MENU_AUTOASSIGN, grow_info );
+            if( !growing_was_started ) {
+                pmenu.addentry( 0, false, MENU_AUTOASSIGN, _( "It is too cold to grow." ) );
             }
-            // Choose fertilizer from list
-            int f_index = 0;
-            if (f_types.size() > 1) {
-                f_names.push_back(_("Cancel"));
-                f_index = menu_vec(false, _("Use which fertilizer?"), f_names) - 1;
-                if (f_index == (int)f_names.size() - 1) {
-                    f_index = -1;
-                }
+            if( is_mature ) {
+                pmenu.addentry( 0, false, MENU_AUTOASSIGN, _( "The plant is mature and will yield fruits." ) );
+            }
+        }
+        pmenu.query();
+
+        int action_index = pmenu.ret;
+
+        // Watering
+        if( action_index == 1 ) {
+            const inventory &crafting_inv = p.crafting_inventory();
+            int water_charges = crafting_inv.charges_of( "water" );
+            if( water_charges == 0 ) {
```

Comment:
No info is given about how much water will be used for watering.

Also no selection of water from which container will be used for watering (that is probably good as it reduces tedium).

Reply:
Water will be used from crafting inventory, same as for cooking.